modules/tensor_mechanics/doc/content/modules/tensor_mechanics/Dynamics.md

```diff
@@ -0,0 +1,128 @@
+
+The degree of damping in the system depends on the coefficients $\zeta$ and $\eta$ as follows:
+\begin{equation}
+\label{eqn:general_rayleigh}
+\xi (\omega) = \frac{\eta}{2} \frac{1}{\omega} + \frac{\zeta}{2} \omega
+\end{equation}
+
+where, $\xi(\omega)$ is the damping ratio of the system as a function of frequency $\omega$. For example, the damping ratio as a function of frequency for $\zeta = 0.0035$ and $\eta = 0.09$ is presented in [fig:rayleigh]. Note that $\omega$ has units of rad/s and $f$ used in the figure has units of Hz.
+
+!media media/tensor_mechanics-rayleigh.png style=width:60%;margin-left:150px;float:center; id=fig:rayleigh caption=Damping ratio as a function of frequency.
+
+To model a constant damping ratio using Rayleigh damping, the aim is to find $\zeta$ and $\eta$ such that the $\xi(f)$ is close to the target damping ratio $\xi_t$, which is a constant value, between the frequency range $[f_1, f_2]$. This can be achieved by minimizing the difference between $\xi_t$ and $\xi(f)$ for all the frequencies between $f_1$ and $f_2$, i.e., if
+
+\begin{equation}
+I = \int_{f_1}^{f_2} \xi_t - \left(\frac{\eta}{2}\frac{1}{\omega} + \frac{\zeta}{2} \omega\right) df
+\end{equation}
+
+Then, $\frac{dI}{d \eta} = 0$ and $\frac{dI}{d \zeta}=0$ results in two equations that are linear in $\eta$ and $\zeta$. Solving these two linear equations simultaneously gives:
+
+\begin{equation}
+\begin{aligned}
+\zeta &= \frac{\xi_t}{2 \pi} \; \frac{3}{(\Delta f)^2} \; \left(f_1 + f_2 - 2 \frac{f_1 f_1}{\Delta f} \; ln \frac{f_2}{f_1}\right) \\
+\eta &= 2 \pi \xi_t \; \frac{f_1 f_2}{\Delta_f} \; \left[ln \frac{f_2}{f_1}\; \left(2 + 6 \frac{f_1 f_2}{(\Delta_f)^2}\right) - \frac{3(f_1 + f_2)}{\Delta_f}\right]
+\end{aligned}
+\end{equation}
+
+A similar method can be used to estimate $\eta$ and $\zeta$ for non-constant damping ratios.
+
+## Implementation and Usage
+
+In the moose framework, the mass and stiffness matrices are not explicitly calculated. Only the residuals are calculated. To get the residual, the equation of motion (with Rayleigh damping and HHT time integration) can be written as:
+
+\begin{equation}
+\rho \ddot{u}(t+\Delta t) + \eta \rho [(1+\alpha)\dot{u}(t + \Delta t)-\alpha \dot{u}(t)] + \zeta  \nabla \cdot [(1+\alpha)\frac{d}{dt}\sigma(t+\Delta t)- \alpha \frac{d}{dt}\sigma(t)] + \nabla \cdot [(1+\alpha) \sigma(t+\Delta t) - \alpha \sigma(t)] = F_{ext}(t+ (1 + \alpha) \Delta t)
+\end{equation}
+
+Here, $\rho$ is the density of the material and $\sigma$ is the stress tensor. The weak form of the above equation is used to get the residuals.
+
+The first two terms to the left ($ \rho \ddot{u}(t+\Delta t) + \eta \rho ((1+\alpha) \dot{u}(t + \Delta t)-\alpha \dot{u}(t)) $) are calculated in the [InertialForce](/InertialForce.md) kernel. $\alpha$, $\beta$, $\gamma$ and $\eta$ need to be passed as input to the InertialForce kernel.
+
+The next two terms to the left involving $\sigma$ are calculated in DynamicStressDivergenceTensors.C.
```

Comment:
Replace this with a link: `[DynamicStressDivergenceTensors](/DynamicStressDivergenceTensors.md)`